controllers/component/component_controller.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions andF
+limitations under the License.
+*/
+
+package component
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/helpers"
+	"github.com/redhat-appstudio/integration-service/loader"
+	"github.com/redhat-appstudio/operator-toolkit/controller"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// Reconciler reconciles a component object
+type Reconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// NewComponentReconciler creates and returns a Reconciler.
+func NewComponentReconciler(client client.Client, logger *logr.Logger, scheme *runtime.Scheme) *Reconciler {
+	return &Reconciler{
+		Client: client,
+		Log:    logger.WithName("integration pipeline"),
+		Scheme: scheme,
+	}
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=components,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=components/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=components/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
```

Comment:
Do we need to also add the entry for the apiGroups: `pipelinesascode.tekton.dev`, like [this](https://github.com/redhat-appstudio/integration-service/blob/fd8631a6931762fdd0863302579d0fb5f17c41db/controllers/buildpipeline/buildpipeline_controller.go#L62)? Because the [check fails on this](https://github.com/redhat-appstudio/integration-service/actions/runs/6153305093/job/16696963235?pr=273#step:13:61), so maybe it would fix the error...

Reply:
I think run `make manifests` before committing the PR can fix that error

Reply:
RBAC failures have been resolved in [1e30d88](https://github.com/redhat-appstudio/integration-service/commit/1e30d880842763338ce6686ac43857701adb2f56)
@hongweiliu17 that is good to know, I will take note of that I ended up doing it manually.

Reply:
On a different topic, I think that we don't need access to secrets for this controller, so this line might not be needed here.

Reply:
Do we need this `configmaps` and `pipelinesascode.tekton.dev`? cc @dirgim  I feel not, but I am not sure.

Reply:
Based on my understanding of the code, I think that they are not needed either, but it would probably be best to confirm by testing.